### PR TITLE
[ADDS] Deletion and option to edit for board configs

### PR DESCRIPTION
**Pull request by MWahhab**
Includes:
- Option to delete configs
- Universal delete confirmation popup
- Fix for initial board rendering issues

### CODE REVIEW

File: resources/js/Pages/Board/components/BoardAmendment.tsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+                    variant: 'destructive',
+                    title: 'Failed to delete board',
+                });
+            },
+        });
+    }
+
+    return (
+        <>
+            <DropdownMenu>
```
Comment (lolmeherti) [inline resources/js/Pages/Board/components/BoardAmendment.tsx]: Can you make this fit the current color scheme?

File: app/Http/Controllers/BoardConfigController.php [hunk 2]
```diff
@@ -88,16 +89,32 @@ public function edit(BoardConfig $boardConfig)
...
+            'columns'   => ['required', 'array', 'min:1', function ($attribute, $value, $fail) {
+                if (count($value) !== count(array_unique($value))) {
+                    $fail('Column names must be unique.');
+                }
+            }],
+            'columns.*' => 'string|min:1|max:255'
+        ]);
+
+        $board->update($validatedReq);
+
```
Comment (lolmeherti) [inline app/Http/Controllers/BoardConfigController.php]: What happens when the edited are no longer named the same? The posts will not be assigned anywhere and they end up being lingering posts in the posts table with nowhere to go.

File: app/Services/BoardService.php [hunk 3]
```diff
@@ -7,13 +7,16 @@
...
-     * Gets each board with their respective posts and each post with their respective comments.
+     * Gets each board with their respective posts and each post with their respective comments. If $boardId is null -
+     * assumed this is the first render and attempts to retrieve the first board config found in the table.
      *
      * @param $boardId
      * @return array|null
      */
     public function getBoardData($boardId = null): ?array
     {
+        BoardConfig::find($boardId) ?? $boardId = null;
```
Comment (lolmeherti) [inline app/Services/BoardService.php]: Remove this line. It is doing an additional query that you absolutely do not need.

File: app/Http/Controllers/BoardConfigController.php [hunk 4]
```diff
@@ -88,16 +89,32 @@ public function edit(BoardConfig $boardConfig)
...
     }
 
     /**
      * Remove the specified resource from storage.
      */
-    public function destroy(BoardConfig $boardConfig)
+    public function destroy(BoardConfig $board):RedirectResponse
     {
-        //
+        $tempId = $board->id;
```
Comment (lolmeherti) [inline app/Http/Controllers/BoardConfigController.php]: Why do we need this temp variable?

File: app/Http/Controllers/BoardConfigController.php [hunk 5]
```diff
@@ -61,13 +62,19 @@ public function store(Request $request): RedirectResponse
...
             'fid_user' => Auth::id()
         ]);
 
-        return redirect()->back()->with('success', 'Board created successfully!');
+//        BoardConfig::create([
+//            'title'    => $validated['title'],
+//            'columns'  => $validated['columns'],
+//            'fid_user' => Auth::id()
+//        ]);
+
```
Comment (lolmeherti) [inline app/Http/Controllers/BoardConfigController.php:72]: What happened here? Why are we committing commented code?

File: resources/js/Pages/Board/components/BoardAmendment.tsx (hunk 1 above)
Comment (MWahhab) [reply resources/js/Pages/Board/components/BoardAmendment.tsx]: Amendment component has been removed

File: app/Http/Controllers/BoardConfigController.php [hunk 6]
```diff
@@ -60,13 +62,13 @@ public function store(Request $request): RedirectResponse
             'columns.*' => 'string|min:1|max:255',
         ]);
 
-        BoardConfig::create([
+        $boardId = DB::table("board_configs")->insertGetId([
```
Comment (lolmeherti) [inline app/Http/Controllers/BoardConfigController.php]: don't swap from the existing way of creating the record

File: app/Http/Controllers/BoardConfigController.php [hunk 7]
```diff
@@ -60,13 +62,13 @@ public function store(Request $request): RedirectResponse
...
             'title'    => $validated['title'],
-            'columns'  => $validated['columns'],
+            'columns'  => json_encode($validated['columns']),
```
Comment (lolmeherti) [inline app/Http/Controllers/BoardConfigController.php]: Let inertia handle this, don't encode by yourself. You're changing core behavior of the boards.

File: resources/js/Pages/Board/BoardLayout.tsx [hunk 8]
```diff
@@ -145,7 +147,8 @@ export function BoardLayout() {
         [setTasks, setColumns]
     );
 
-    const [selectedTask, setSelectedTask]         = useState(null);
+    // State for selected task and dialog visibility
```
Comment (lolmeherti) [inline resources/js/Pages/Board/BoardLayout.tsx]: Why'd you add this obvious comment?

**Pull request merged**